alert-database/src/main/java/com/synopsys/integration/alert/database/api/StaticJobAccessor.java

```diff
@@ -178,6 +183,42 @@ public StaticJobAccessor(
         return getMatchingEnabledJobs(() -> distributionJobRepository.findMatchingEnabledJob(providerConfigId, notificationType.name()));
     }
 
+    @Override
+    public List<FilteredDistributionJobModel> getMatchingEnabledJobs(FilteredDistributionJobRequestModel filteredDistributionJobRequestModel) {
+        FrequencyType frequencyType = filteredDistributionJobRequestModel.getFrequencyType();
+        NotificationType notificationType = filteredDistributionJobRequestModel.getNotificationType();
+
+        String projectName = filteredDistributionJobRequestModel.getProjectName();
+
+        List<String> policyNames = filteredDistributionJobRequestModel.getPolicyNames();
+        List<String> vulnerabilitySeverities = filteredDistributionJobRequestModel.getVulnerabilitySeverities();
+
+        List<DistributionJobEntity> distributionJobEntities;
+        if (!policyNames.isEmpty()) {
+            distributionJobEntities = distributionJobRepository.findMatchingEnabledJobsWithPolicyNames(frequencyType.name(), notificationType.name(), projectName, policyNames);
+        } else if (!vulnerabilitySeverities.isEmpty()) {
```

Comment:
There's no reason this would be mutually exclusive with policyNames.

Reply:
The filtering is done per notification and thus will only have one or the other.